[BE] Test cases for login

MihaiDruta opened the pull request:
[BE] Test cases for login
https://trello.com/c/RsPpY9pu/107-be-test-cases-for-login

Added test cases for login

MihaiDruta committed:
[BE] Test cases for login
https://trello.com/c/RsPpY9pu/107-be-test-cases-for-login

Added test cases for login

MihaiDruta committed:
[BE] Test cases for login
https://trello.com/c/RsPpY9pu/107-be-test-cases-for-login

Verified if the returned token has the expected payload

MihaiDruta committed:
[BE] Test cases for login
https://trello.com/c/RsPpY9pu/107-be-test-cases-for-login

Added return type to LoginTest.php functions

MihaiDruta committed:
[BE] Test cases for login
https://trello.com/c/RsPpY9pu/107-be-test-cases-for-login

Added more activity types in fixtures

MihaiDruta committed:
[BE] Test cases for login
https://trello.com/c/RsPpY9pu/107-be-test-cases-for-login

MihaiDruta committed:
[BE] Test cases for login
https://trello.com/c/RsPpY9pu/107-be-test-cases-for-login

MihaiDruta committed:
[BE] Test cases for login
https://trello.com/c/RsPpY9pu/107-be-test-cases-for-login

MihaiDruta committed:
[BE] Test cases for login
https://trello.com/c/RsPpY9pu/107-be-test-cases-for-login

MihaiDruta committed:
[BE] Test cases for login
https://trello.com/c/RsPpY9pu/107-be-test-cases-for-login

Generated public and private keys in travis.yml configuration file

SpartakusMd commented on .env.test:
```diff
@@ -2,3 +2,9 @@
+JWT_SECRET_KEY=%kernel.project_dir%/config/jwt/private.pem
+JWT_PUBLIC_KEY=%kernel.project_dir%/config/jwt/public.pem
+JWT_PASSPHRASE=passphrase
+###< lexik/jwt-authentication-bundle ###
```
Leave an empty line at the end.

SpartakusMd commented on src/DataFixtures/TypeFixtures.php:
```diff
@@ -12,7 +12,13 @@ class TypeFixtures extends Fixture
     public const TYPES = [
         'dt estimation',
```
Make the words capitalized where needed. The shouldn't be all lowercase.

SpartakusMd commented on tests/LoginTest.php:
```diff
@@ -0,0 +1,109 @@
+
+    private function getUsernameRoles($token): array
+    {
+        $decodedToken = $this->tokenDecode($token);
```
Use `Lexik\Bundle\JWTAuthenticationBundle\Encoder\JWTEncoderInterface::decode()`

SpartakusMd reviewed (changes requested):
Travis is still failing.

MihaiDruta committed:
[BE] Test cases for login
https://trello.com/c/RsPpY9pu/107-be-test-cases-for-login

Using JWTEncoderInterface for token decoding instead of doing this manually

SpartakusMd commented on src/DataFixtures/ActivityFixtures.php:
```diff
@@ -31,7 +31,7 @@ public function load(ObjectManager $manager): void
 
             /** @var ActivityType $type */
-            $type = $this->getReference('type_' . TypeFixtures::TYPES[rand(0, 2)]);
+            $type = $this->getReference('type_' . TypeFixtures::TYPES[rand(0, 8)]);
```
8 will be undefined. The count starts from 0 not 1.

MihaiDruta committed:
[BE] Test cases for login
https://trello.com/c/RsPpY9pu/107-be-test-cases-for-login
